asap/rough_align/fit_multiple_solves.py

```diff
@@ -0,0 +1,367 @@
+        "montage_pt_weight": 1.0
+    }
+
+    tps_matches = copy_matches(new_matches)  # any transform_apply modifies matches.... bummer.
+
+    tps_create_CRS_A_input = (
+        tpsadded_aff_result_rts, tps_matches, tps_transform_name,
+        tps_transform_apply, tps_regularization_dict, tps_matrix_assembly_dict, tps_order, tps_fullsize)
+    return tps_create_CRS_A_input,tps_regularization_dict
+
+
+class FitMultipleSolves(argschema.ArgSchemaParser):
+    default_schema = FitMultipleSolvesSchema
+    default_output_schema = FitMultipleSolvesOutputSchema
+
+    def apply_transform(self,new_matches,rts,app):
+        app_solver = {'rigid': rigid_solve,
+            'translate': translation_solve,
+            'affine': affine_solve,
+            'tps': thin_plate_spline_solve}
+        app_input,app_reg_dict = app_solver[app](rts,new_matches)
+        app_fr, app_draft_resolvedtiles = bigfeta.bigfeta.create_CSR_A_fromobjects(*app_input, return_draft_resolvedtiles=True)
+        app_result_rts = copy.deepcopy(app_draft_resolvedtiles)
+        app_reg = scipy.sparse.diags(
+            [numpy.concatenate([ts.tforms[-1].regularization(app_reg_dict) for ts in app_draft_resolvedtiles.tilespecs])],
+            [0], format="csr")
+        app_sol = bigfeta.solve.solve(app_fr["A"], app_fr["weights"], app_reg, app_fr["x"], app_fr["rhs"])
+        bigfeta.utils.update_tilespecs(app_result_rts, app_sol["x"])
+        return app_result_rts
+
+    def save_transform(self,render,rts,app):
+        outstacks = {'rigid':self.args['rigid_output_stack']['name'],
+                    'translate': self.args['translation_output_stack']['name'],
+                    'affine': self.args['affine_output_stack']['name'],
+                    'tps': self.args['thin_plate_output_stack']['name']}
+        app_outstack=outstacks[app]
+        renderapi.stack.create_stack(app_outstack,render=render)
+        renderapi.resolvedtiles.put_tilespecs(app_outstack, rts,render=render))
+        return None
+        
+    def run(self):
+        r_in = renderapi.connect(**self.args['input_stack'])
+        r_pm = renderapi.connect(**self.args['pointmatch_collection'])
+        r_rot = renderapi.connect(**self.args['rigid_output_stack'])
+        r_aff = renderapi.connect(**self.args['affine_output_stack'])
+        r_tps = renderapi.connect(**self.args['thin_plate_output_stack'])
+        if self.args['translation_output_stack'] is not None:
+            r_trans = renderapi.connect(**self.args['translation_output_stack'])
+            do_translate=True
+        else:
+            do_translate = False
+        allZ = [int(z) for z in renderapi.stack.get_z_values_for_stack(self.args['input_stack']['name'], render=r_in)]
+        if self.args['minZ'] is None:
+            minZ = allZ[0]
+        else:
+            minZ = self.args['minZ']
+        if self.args['maxZ'] is None:
+            maxZ = allZ[-1]
+        else:
+            maxZ = self.args['maxZ']
```

Comment:
```suggestion
        minZ = (allZ[0] if self.args["minZ"] is None else max(self.args["minZ"], allZ[0]))
        maxZ = (allZ[-1] if self.args["maxZ"] is None else min(self.args["maxZ"], allZ[-1]))
```
This representation should clip the minZ and maxZ to values that exist in the stack, which could be better for e.g. retrieving pointmatches.